error[E0521]: borrowed data escapes outside of method
  --> $DIR/issue-16683.rs:4:9
   |
LL |     fn b(&self) {
   |          ----- `self` is a reference that is only valid in the method body
LL |         self.a();
   |         ^^^^^^^^ `self` escapes the method body here

error: aborting due to previous error

